File: src/hi/hi.go
package hi

import (
	"shylinux.com/x/ice"
	"shylinux.com/x/icebergs/base/ctx"
)

type hi struct {
	ice.Zone

	list string `name:"list zone id auto insert" help:"示例"`
}

func (s hi) Show(m *ice.Message, arg ...string) {
	m.Echo("hello world").StatusTime()
}

func (s hi) List(m *ice.Message, arg ...string) {
	// ctx.Display(m, "hi.js")
	// ctx.Display(m, "/require/shylinux.com/x/contexts@v2.9.2/src/hi/hi.js")
	// ctx.Display(m, "http://localhost:9020/require/shylinux.com/x/contexts@v2.9.2/src/hi/hi.js")
	ctx.Display(m, "https://shylinux.com/x/contexts@v2.9.2/src/hi/hi.js?content=what")
	s.Zone.ListPage(m, arg...)
}

func init() { ice.Cmd("web.code.hi.hi", hi{}) }

File: src/hi/hi.js
Volcanos(chat.ONIMPORT, {_init: function(can, msg) {
	msg.Defer(function() { msg.Dump() })
	msg.Echo("hello world\n"+(can.Conf("content")||""))
	can.require(["he.js"], function() { }, function() { })
	// can.require(["/require/shylinux.com/x/contexts@v2.9.2/src/hi/he.js"], function() { }, function() { })
	// can.require(["http://localhost:9020/require/shylinux.com/x/contexts@v2.9.2/src/hi/he.js"], function() { }, function() { })
	// can.require(["https://shylinux.com/x/contexts@v2.9.2/src/hi/he.js?content=what"], function() { }, function() { })
}})
